toolkit/tools/scheduler/scheduler.go

```diff
@@ -387,6 +388,19 @@ func buildAllNodes(stopOnFailure, isGraphOptimized, canUseCache bool, packagesTo
 					}
 				}
 
+				if res.Node.Type == pkggraph.TypeBuild && res.WasDelta {
+					logger.Log.Tracef("This is a delta result, update the graph with the new delta files for '%v'.", res.Node)
+					// We will need to update the graph with paths to any delta files that were actually rebuilt.
+					err = setAssociatedDeltaPaths(res, res.BuiltFiles, pkgGraph, graphMutex)
+					if err != nil {
+						// Failures to manipulate the graph are fatal.
+						err = fmt.Errorf("error setting delta paths for ancillary nodes: %w", err)
+						logger.Log.Error(err)
+						stopBuilding = true
+						stopBuild(channels, buildState)
```

Comment:
>stopBuild(channels, buildState) [](http://example.com/codeflow?start=6&length=31)

This is a third `stopBuild(channels, buildState)`. Do you think we could just move all of them into one call right below the `if !stopBuilding` section? We're sure not to change the current logic this way. I'm curious if we could also just move it under the existing `if stopBuilding` a bit later in the code.

Reply:
Moved to a single location